marketplaceItems/questionApp/appResources/appData/question_app.js

```diff
@@ -0,0 +1,238 @@
+//
+//  Created by Rebecca Stankus on 02/21/19
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* global  Audio, Camera, Controller, Entities, HMD, Messages, MyAvatar, Quat, Script, Settings, SoundCache, 
+Tablet, Users, Vec3, Window */
+
+(function() {
+
+    // *************************************
+    // START UTILITY FUNCTIONS
+    // *************************************
+
+    var PickRayController = Script.require('./resources/modules/pickRayController.js?' + Date.now());
+    var pickRayController = new PickRayController();
+
+    /* PLAY A SOUND: Plays the specified sound at the position of the user's Avatar using the volume and playback 
+    mode requested. */
+    var injector;
+    function playSound(sound, volume, position, localOnly, loop){
+        if (sound.downloaded) {
+            if (injector) {
+                injector.stop();
+                injector = null;
+            }
+            injector = Audio.playSound(sound, {
+                position: position,
+                volume: volume,
+                localOnly: localOnly,
+                loop: loop
+            });
+        }
+    }
+
+    // *************************************
+    // END UTILITY FUNCTIONS
+    // *************************************
+
+    /* CREATE A QUESTION MARK: Checks that question Mark does not already exist, then calculates position above 
+    avatar's head and creates a question mark entity there */
+    var QUESTION_MARK_Y_OFFSET_RATIO_FROM_HEAD_TOP = 0.3;
+    var QUESTION_MARK_Y_OFFSET_RATIO_FROM_HEAD = 0.4;
+    var QUESTION_MARK_Y_OFFSET_RATIO_FROM_HIPS = 0.6;
+    var GROWTH_INTERVAL_MS = 1000; // takes 4 min 38 sec to reach height of 0.8 from 0.2 at ratio:1.005, interval: 1000
+    var MAX_HEIGHT_M = 0.8;
+    var GROWTH_RATIO = 1.1;
+    var questionMark;
+    var questionMarkMaterial;
+    var growingInterval;
+    function createQuestionMark() {
+        if (questionMark) {
+            return;
+        }
+        var avatarHeight = MyAvatar.getHeight();
+        var parentJointIndex = MyAvatar.getJointIndex("HeadTop_End");
+        var offsetRatio = QUESTION_MARK_Y_OFFSET_RATIO_FROM_HEAD_TOP;
+        print("USING HEAD TOP JOINT");
+        if (parentJointIndex === -1) {
+            print("NOPE>>>USING HEAD JOINT");
+            parentJointIndex = MyAvatar.getJointIndex("Head");
+            offsetRatio = QUESTION_MARK_Y_OFFSET_RATIO_FROM_HEAD;
+        }
+        if (parentJointIndex === -1) {
+            parentJointIndex = MyAvatar.getJointIndex("Hips");
+            offsetRatio = QUESTION_MARK_Y_OFFSET_RATIO_FROM_HIPS;
+            print("ERROR: Falling back to hips joint as head could not be found");
+        }
+        var questionMarkLocalYPosition = avatarHeight * offsetRatio;
+        questionMark = Entities.addEntity({
+            name: "Question App Mark",
```

Comment:
please pull out the static properties into their own `var` above this function

Reply:
Also, this name is referenced more than once in this file (line 187). please move the entity name into its own var as well, then update line 187, too.